Move buffer and window classes to correct files

jmercouris opened the pull request:
This is with regards to https://github.com/atlas-engineer/nyxt/issues/581

In a subsequent pull request, the methods can be moved. This is the minimum necessary to move the classes.

Ambrevar commented on source/package.lisp:
```diff
@@ -27,4 +27,12 @@ It's recommended to use this package in the Nyxt configuration file, instead of
+  (:import-from #:keymap #:define-key #:define-scheme)
+  (:import-from #:serapeum #:export-always)
+  (:export :minibuffer-mode)
+  (:documentation "Mode for minibuffer"))
```
Can you add a comment explaining why this mode must be here?
Otherwise the reader may not understand why this is inconsistently declared here, unlike other modes.

jmercouris replied:
will do!

Ambrevar commented on source/minibuffer-mode.lisp:
```diff
@@ -63,6 +58,58 @@
+              (kill-whole-line minibuffer)
+              (insert minibuffer (str:concat (shortcut engine) " "))))))))
+    (t
+     (insert-suggestion minibuffer))))
```
We have a lot of search-engine related code.  What about moving it to a new search-engines.lisp file?

jmercouris replied:
we can do that, it is outside of the scope of this pull request, so I will do it in a separate commit

Ambrevar commented:
Thank you so much for taking a shot at this, it's been itching me for way too long! :)

Looking good beside the search-engine suggestion.

jmercouris commented:
Yay!

Ambrevar commented:
> we can do that, it is outside of the scope of this pull request, so I will do it in a separate commit

But you've already moved them.
The downside of moving code around too many times is that it's
harder to track the Git history of the related code.

jmercouris commented:
It sure is, luckily we are only going to do this once :-D

jmercouris commented:
Ah, I just understood what you meant by "I've already moved them"! True, but I moved them for compilation purposes, which I think does justify a discrete commit.
